pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -170,19 +171,42 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 			}
 		}
 		// fire instance handles for workload
-		ep := NewEndpointBuilder(pc.c, pod).buildIstioEndpoint(ip, 0, "", model.AlwaysDiscoverable)
-		workloadInstance := &model.WorkloadInstance{
-			Name:      pod.Name,
-			Namespace: pod.Namespace,
-			Kind:      model.PodKind,
-			Endpoint:  ep,
-			PortMap:   getPortMap(pod),
+		if len(pc.c.handlers.GetWorkloadHandlers()) > 0 {
+			ep := NewEndpointBuilder(pc.c, pod).buildIstioEndpoint(ip, 0, "", model.AlwaysDiscoverable)
+			workloadInstance := &model.WorkloadInstance{
+				Name:      pod.Name,
+				Namespace: pod.Namespace,
+				Kind:      model.PodKind,
+				Endpoint:  ep,
+				PortMap:   getPortMap(pod),
+			}
+			pc.c.handlers.NotifyWorkloadHandlers(workloadInstance, ev)
 		}
-		pc.c.handlers.NotifyWorkloadHandlers(workloadInstance, ev)
 	}
 	return nil
 }
 
+// checkPodLabelChange fires off eds update if pod label changes
+func (pc *PodCache) checkPodLabelChange(old, cur interface{}) bool {
+	oldPod, ok := old.(*v1.Pod)
+	if !ok {
+		return true
+	}
+	curPod, ok := cur.(*v1.Pod)
+	if !ok {
+		return true
+	}
+	if !labels.Instance(oldPod.Labels).Equals(curPod.Labels) {
+		// fire off eds update on pod label change
+		if services, err := getPodServices(pc.c.serviceLister, curPod); err == nil {
+			for _, svc := range services {
+				pc.queueEndpointEvent(kube.KeyFunc(svc.Name, svc.Namespace))
```

Comment:
Not to use `queueEndpointEvent`, it is aimed to do resync when pod comes later than endpoint

Reply:
Does it really matter? Please correct me if I'm wrong

Reply:
It is confusing, or you have to update the comment In L43

Reply:
OK, thanks for reminding

Reply:
I've added a comment for `queueEndpointEvent`